### PR TITLE
[IMP] mrp: stop creating return on done transfer

### PR DESCRIPTION
**Pull request by temo-odoo**
Before commit:
================
In MO if a line is deleted or line has its to consume quantity reduced so in done state transfers return is creating.

After commit:
==============
Stopped creating the return on done state transfer when quantity is reduced or a line is deleted.

task: 3283010

### CODE REVIEW

**Comment by Whenrow** [comment]
@temo-odoo could you resolve the conflict and write a test for this feature ? Thank you

**Comment by temo-odoo** [comment]
Hello @Whenrow 
Conflicts are resolved and added test case.
Thanks

File: addons/mrp/models/stock_move.py [hunk 1]
```diff
@@ -313,6 +313,8 @@ def _run_procurement(self, old_qties=False):
         to_assign = self.env['stock.move']
         self._adjust_procure_method()
         for move in self:
+            if float_compare(move.product_uom_qty - old_qties[move.id], 0, precision_rounding=move.product_uom.rounding) < 0 and not move.move_orig_ids.filtered(lambda m: m.state != 'done'):
```
[the remark this answers is not in the document]
Comment (temo-odoo) [reply addons/mrp/models/stock_move.py]: Hello @clesgow 
Your suggested changes are done. Please check once.
Thanks.

**Comment by Whenrow** [comment]
Hello. Sorry for the long delay. As the 17.0 freeze has been done, could you retarget your PR and branch on 17.0 version ? Thank you

**Comment by temo-odoo** [comment]
Hello @Whenrow 

I have retargeted the PR and branch to the 17.0 version. Some js tests have failed, and I don't think it's due to my code

Thank you.

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**